sql-plugin/src/main/scala/org/apache/spark/sql/rapids/execution/GpuBroadcastExchangeExec.scala

```diff
@@ -87,18 +88,19 @@ object SerializedHostTableUtils extends Arm {
   }
 }
 
+// scalastyle:off no.finalize
 @SerialVersionUID(100L)
 class SerializeConcatHostBuffersDeserializeBatch(
     data: Array[SerializeBatchDeserializeHostBuffer],
     output: Seq[Attribute])
-  extends Serializable with Arm with AutoCloseable {
+  extends Serializable with Arm with AutoCloseable with Logging {
   @transient private var dataTypes = output.map(_.dataType).toArray
   @transient private var headers = data.map(_.header)
   @transient private var buffers = data.map(_.buffer)
-  @transient private var batchInternal: ColumnarBatch = null
```

Comment:
it would be really nice, although I know this is not something you did, to add a comment above `batchInternal` saying that on the host it stays null (or `None` if you choose to undertake the `Option` mission)